Skill/app/src/main/java/com/skilltradiez/skilltraderz/TradeList.java

```diff
@@ -156,10 +179,40 @@ public boolean commit(UserDatabase userDB) {
             if (!trade.commit(userDB))
                 return false;
         }
+
+        /** There might be a bug here, I did this delete trade todo... /signed Cole **/
         newTrades.clear();
         for (ID tradeId : deletedTrades) {
-            //TODO Delete trade
+            Trade trade = DatabaseController.getTradeByID(tradeId);
+            User currentUser = DatabaseController.getAccountByUserID(getOwnerID());
+            User tradePartner = DatabaseController.getAccountByUserID(trade.getHalf2().getUser());
+
+            //If given a null just break through it.
+            if (tradeId == null){
```

Comment:
I agree with this, I am trying to fit in an exception being thrown. Having a very bad struggle implementing it.